opm/grid/cpgrid/Entity.hpp

```diff
@@ -566,6 +574,38 @@ Dune::cpgrid::Entity<0> Dune::cpgrid::Entity<codim>::getOrigin() const
     }
 }
 
+template<int codim>
+Dune::cpgrid::Entity<0> Dune::cpgrid::Entity<codim>::getLgrElem() const
+{
+    // Check that the element belongs to the leaf grid view
+    if (!(pgrid_ -> leaf_to_level_cells_.empty())) // entity on the LeafGridView
+    {
+        const auto entityLevel = this -> level(); // pgrid_->leaf_to_level_cells_[this->index()][0]
+        const int& entityLgrIdx = pgrid_->leaf_to_level_cells_[this->index()][1]; // leaf_to_level_cells_ [leaf idx] = {level, cell idx}
+        const auto& lgr_grid = (*(pgrid_ -> level_data_ptr_))[entityLevel].get();
+        return Dune::cpgrid::Entity<0>( *lgr_grid, entityLgrIdx, true);
+    }
+    else {
+        throw std::invalid_argument("The entity provided does not belong to the leaf grid view. ");
```

Comment:
Would it make sense to return `*this` here? 

My understanding is that we return the entity of a level grid view that is equivalent to he one on the level. If there is no refinement, then they are the same

Reply:
With my current knowledge this comment is just dumb. Resolving this one.